Redesign UI

ztimms73 opened the pull request:
The work on changing the interface is complete. The application was tested on API 21, 23, 29 and no bugs were found in the interface. Here is a table with screenshots of the old and new versions for comparison. 

| Old | New | 
| ------------- | ------------- |
| ![Screenshot_1623880672](https://user-images.githubusercontent.com/61558546/122301541-79cc0080-cf09-11eb-9ce0-27a16f6452dd.png) | ![Screenshot_1623880963](https://user-images.githubusercontent.com/61558546/122301584-8c463a00-cf09-11eb-97f0-7e9cfd41c4c1.png) |
| ![Screenshot_1623880680](https://user-images.githubusercontent.com/61558546/122301720-c0215f80-cf09-11eb-863b-de6635278523.png) | ![Screenshot_1623880971](https://user-images.githubusercontent.com/61558546/122301747-ca435e00-cf09-11eb-809d-6b105e42eaa3.png) |
| ![Screenshot_1623880692](https://user-images.githubusercontent.com/61558546/122301831-ea731d00-cf09-11eb-9e2b-120382490add.png) | ![Screenshot_1623880977](https://user-images.githubusercontent.com/61558546/122301861-f52db200-cf09-11eb-8968-e299725fddaf.png) |
| ![Screenshot_1623880696](https://user-images.githubusercontent.com/61558546/122301928-10002680-cf0a-11eb-8204-44aaa2510c3e.png) | ![Screenshot_1623880982](https://user-images.githubusercontent.com/61558546/122301946-18586180-cf0a-11eb-864a-0c8893c1b514.png) |
| ![Screenshot_1623880733](https://user-images.githubusercontent.com/61558546/122302036-3a51e400-cf0a-11eb-92a3-80278717ae7b.png) | ![Screenshot_1623881011](https://user-images.githubusercontent.com/61558546/122302070-45a50f80-cf0a-11eb-9fae-a0d02808765a.png) |
| ![Screenshot_1623880737](https://user-images.githubusercontent.com/61558546/122302103-52296800-cf0a-11eb-886b-77186f2d4a4d.png) | ![Screenshot_1623881015](https://user-images.githubusercontent.com/61558546/122302121-58b7df80-cf0a-11eb-9df6-8bedf67fd09d.png) |
| ![Screenshot_1623880744](https://user-images.githubusercontent.com/61558546/122302274-8b61d800-cf0a-11eb-93de-38fcd51d2443.png) | ![Screenshot_1623881022](https://user-images.githubusercontent.com/61558546/122302299-93217c80-cf0a-11eb-88ea-0e545a8b612d.png) |
| ![Screenshot_1623880775](https://user-images.githubusercontent.com/61558546/122302445-ccf28300-cf0a-11eb-9868-ca2406d1e35a.png) | ![Screenshot_1623881038](https://user-images.githubusercontent.com/61558546/122302477-d67beb00-cf0a-11eb-8330-d9b6b2b95e2b.png) |

| Local manga | Remote manga | 
| ------------- | ------------- |
| ![Screenshot_1623881038](https://user-images.githubusercontent.com/61558546/122302789-45594400-cf0b-11eb-87a3-857db2bc66ab.png) | ![Screenshot_1623881105](https://user-images.githubusercontent.com/61558546/122302819-4d18e880-cf0b-11eb-89b2-2b593af45d67.png) |

| Tablet portrait (Nexus 7 (2013)) | Tablet landscape (Nexus 7 (2013)) | 
| ------------- | ------------- |
| ![Screenshot_1623881439](https://user-images.githubusercontent.com/61558546/122303077-a2ed9080-cf0b-11eb-891d-11f007be70ab.png) | ![Screenshot_1623881425](https://user-images.githubusercontent.com/61558546/122303196-d3cdc580-cf0b-11eb-9977-efef8dccac08.png) |

nv95 commented:
Looks very good, I'll review it when I'm at home. Also, what about new search ui?

ztimms73 commented:
@nv95, are you talking about the SearchUI that you made 2 weeks ago, or about the MaterialSearchView that I tried to implement? If about MaterialSearchView, then I need to think about how to replace the Toolbar with this. MaterialSearchView has a flaw, it can't inflate menu items, so I modified it a bit and added `OnMenuItemClickListener`. Also, I need to think about how to link it with SearchUI. You can read more about MaterialSearchView [here](https://github.com/lapism/search).

nv95 commented:
Ok, let's merge this firstly. 
Maybe it will be better to implement ui like MaterialSearchView without this library using ActionMenuView (from androidx) for option menu

nv95 commented on app/src/main/java/org/koitharu/kotatsu/base/ui/widgets/ChipsView.kt:
```diff
@@ -4,17 +4,18 @@ import android.content.Context
 class ChipsView @JvmOverloads constructor(
 	context: Context,
-	attrs: AttributeSet? = null,
-	defStyleAttr: Int = R.attr.chipGroupStyle
```
Why you remove this?

ztimms73 replied:
Oh, I tried to make the attributes of the chip group styling, I temporarily removed it, but I forgot to revert it

nv95 commented on app/src/main/java/org/koitharu/kotatsu/base/ui/widgets/search/util/SearchUtils.kt:
```diff
@@ -0,0 +1,37 @@
+	fun isVoiceSearchAvailable(context: Context): Boolean {
+		val pm = context.packageManager
+		val activities =
+			pm.queryIntentActivities(Intent(RecognizerIntent.ACTION_RECOGNIZE_SPEECH), 0)
```
We need to add something to manifest to accomplish restrictions in Android 11?

ztimms73 replied:
Yep, the manifest must contain the following lines
```
<uses-permission android:name="android.permission.RECORD_AUDIO" />

<queries>
	<intent>
		<action android:name="android.speech.RecognitionService" />
	</intent>
</queries>
```

ztimms73 replied:
But, does the app need this permission if it doesn't have a speech-to-text feature? Still, is MaterialSearchView planned to be added?

nv95 replied:
Is RECORD_AUDIO necessary?

nv95 commented on app/src/main/java/org/koitharu/kotatsu/details/ui/DetailsFragment.kt:
```diff
@@ -60,17 +60,41 @@ class DetailsFragment : BaseFragment<FragmentDetailsBinding>(), View.OnClickList
-			imageViewFavourite.setOnClickListener(this@DetailsFragment)
+			val file = manga.url.toUri().toFileOrNull()
+			if (file != null) {
+				val size = file.length()
```
Blocking call, it will be better to do this async

nv95 commented on app/src/main/res/values/styles.xml:
```diff
@@ -1,43 +1,82 @@
+	<!-- TextAppearance -->
+
+	<style name="TextAppearance.Kotatsu.ToolbarTitle" parent="@style/TextAppearance.Widget.AppCompat.Toolbar.Title">
+		<item name="android:textSize" tools:ignore="SpUsage">20dp</item>
```
Why `dp` instead of `sp`?

ztimms73 replied:
Whoops, sorry, I didn't notice this, in the Google I/O repository, for some reason, `dp` is presented, not `sp`, which is really weird

nv95 commented on app/src/main/java/org/koitharu/kotatsu/details/ui/DetailsFragment.kt:
```diff
@@ -60,17 +60,41 @@ class DetailsFragment : BaseFragment<FragmentDetailsBinding>(), View.OnClickList
-			imageViewFavourite.setOnClickListener(this@DetailsFragment)
+			val file = manga.url.toUri().toFileOrNull()
+			if (file != null) {
+				val size = file.length()
+				textViewSize.text = FileSizeUtils.formatBytes(requireContext(), size)
+				sizeContainer.isVisible = true
```
```suggestion
				viewLifecycleScope.launch {
				    val size = withContext(Dispatchers.IO) {
				        file.length()
				    }
				    textViewSize.text = FileSizeUtils.formatBytes(requireContext(), size)
				}
				sizeContainer.isVisible = true
```

ztimms73 commented:
How can I resolve conflicts?
